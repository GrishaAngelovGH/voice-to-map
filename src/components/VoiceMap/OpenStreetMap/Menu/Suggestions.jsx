import React from 'react'
import { Fragment } from 'react'

import Dialog from '@mui/material/Dialog'
import AppBar from '@mui/material/AppBar'
import Toolbar from '@mui/material/Toolbar'
import IconButton from '@mui/material/IconButton'
import Typography from '@mui/material/Typography'
import List from '@mui/material/List'
import ListItem from '@mui/material/ListItem'
import ListItemText from '@mui/material/ListItemText'
import Divider from '@mui/material/Divider'
import Slide from '@mui/material/Slide'

import RecordVoiceOverIcon from '@mui/icons-material/RecordVoiceOver'
import CloseIcon from '@mui/icons-material/Close'

const Transition = React.forwardRef(function Transition(props, ref) {
	return <Slide direction='up' ref={ref} {...props} />
})

const suggestions = [
	'Bulgaria Sofia National Palace of Culture',
	'Bulgaria Sofia Zoo',
	'Bulgaria Sofia National History Museum',
	'London Palace of Westminster',
	'London Buckingham Palace',
	'London Piccadilly Circus',
	'London Big Ben',
	'Tower Bridge London',
	'Germany Brandenburg Gate',
	'France Eiffel Tower'
]

const Suggestions = ({ onClose }) => (
	<Dialog fullScreen={true} open={true} TransitionComponent={Transition}>
		<AppBar sx={{ position: 'sticky' }} color='secondary'>
			<Toolbar>
				<IconButton
					edge='start'
					color='inherit'
					onClick={onClose}
					aria-label='close'
				>
					<CloseIcon />
				</IconButton>
				<Typography sx={{ ml: 2, flex: 1 }} variant='h6' component='div'>
					Voice Search Suggestions
				</Typography>
			</Toolbar>
		</AppBar>
		<List>
			{
				suggestions.map((v, i) => (
					<Fragment key={i}>
						<ListItem button>
							<Fragment >
								<ListItemText primary={<span><RecordVoiceOverIcon /> {v}</span>} />
							</Fragment>
						</ListItem>
						<Divider />
					</Fragment>
				))
			}
		</List>
	</Dialog>
)

export default Suggestions
