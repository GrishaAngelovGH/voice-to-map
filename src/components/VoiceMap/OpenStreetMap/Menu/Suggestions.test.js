import { render } from '@testing-library/react'

import Suggestions from './Suggestions'

test('should render Suggestions component', () => {
    const view = render(<Suggestions />)

    expect(view).toMatchSnapshot()
})