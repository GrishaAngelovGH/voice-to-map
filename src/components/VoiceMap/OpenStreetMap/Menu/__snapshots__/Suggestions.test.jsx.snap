// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Suggestions component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 1024px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    />
    <div
      class="MuiDialog-root MuiModal-root css-1424xw8-MuiModal-root-MuiDialog-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-1cennmq-MuiBackdrop-root-MuiDialog-backdrop"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper css-19do60a-MuiDialog-container"
        role="presentation"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-labelledby=":r0:"
          aria-modal="true"
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullScreen css-l09ud3-MuiPaper-root-MuiDialog-paper"
          role="dialog"
          style="--Paper-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);"
        >
          <header
            class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorSecondary MuiAppBar-positionFixed mui-fixed css-149qkrz-MuiPaper-root-MuiAppBar-root"
            style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12); padding-right: 1024px;"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ygil4i-MuiToolbar-root"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1fytndg-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="CloseIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </button>
              <div
                class="MuiTypography-root MuiTypography-h6 css-ybxm6-MuiTypography-root"
              >
                Voice Search Suggestions
              </div>
            </div>
          </header>
          <ul
            class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    Bulgaria Sofia National Palace of Culture
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    Bulgaria Sofia Zoo
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    Bulgaria Sofia National History Museum
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    London Palace of Westminster
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    London Buckingham Palace
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    London Piccadilly Circus
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    London Big Ben
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    Tower Bridge London
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    Germany Brandenburg Gate
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-cfd5wn-MuiListItem-root"
            >
              <div
                class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                >
                  <span>
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                      data-testid="RecordVoiceOverIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <circle
                        cx="9"
                        cy="9"
                        r="4"
                      />
                      <path
                        d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4m7.76-9.64-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27M20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14"
                      />
                    </svg>
                     
                    France Eiffel Tower
                  </span>
                </span>
              </div>
            </li>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
            />
          </ul>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
